src/components/toolbar/toolbar.js

```diff
@@ -89,6 +94,48 @@ function mdToolbarDirective($$rAF, $mdConstant, $mdUtil, $mdTheming, $animate )
         // to scroll events.
         scope.$on('$mdContentLoaded', onMdContentLoad);
 
+        // If the toolbar is used inside an ng-if statement, we may miss the
+        // $mdContentLoaded event, so we attempt to fake it if we have a
+        // md-content close enough.
+        scope.$watch('scrollShrink', function(scrollShrink) {
+          var closestContent = element.parent().find('md-content');
+
+          // If we have a content element, fake the call; this might still fail
+          // if the content element isn't a sibling of the toolbar
+          if (!contentElement && closestContent.length) {
+            onMdContentLoad(null, closestContent);
+          }
+
+          if (contentElement) {
+            // Disable only if the attribute's expression evaluates to false
+            if (scrollShrink === false) {
+              disableScrollShrink();
+            } else {
+              enableScrollShrink();
+            }
+          }
+        });
+
+        // If the scope is destroyed (which could happen with ng-if), make sure
+        // to disable scroll shrinking again
+        scope.$on('$destroy', function() {
+          disableScrollShrink();
+        });
+
+        function enableScrollShrink() {
+          contentElement.on('scroll', debouncedContentScroll);
+          contentElement.attr('scroll-shrink', 'true');
+
+          $$rAF(updateToolbarHeight);
+        }
+
+        function disableScrollShrink() {
+          contentElement.off('scroll', debouncedContentScroll);
+          contentElement.attr('scroll-shrink', 'false');
+
+          $$rAF(updateToolbarHeight);
+        }
+
         function onMdContentLoad($event, newContentEl) {
           // Toolbar and content must be siblings
           if (element.parent()[0] === newContentEl.parent()[0]) {
```

Comment:
In what case will it do so? All of my testing showed that it always passed the `newContentEl`.